extensions/amp-form/0.1/amp-form.js

```diff
@@ -155,14 +168,40 @@ export class AmpForm {
     /** @const @private {!./form-validators.FormValidator} */
     this.validator_ = getFormValidator(this.form_);
 
-    // TODO(mkhatib, #6927): Wait for amp-selector to finish loading if the current form
-    // is using it.
-    this.actions_.installActionHandler(
-        this.form_, this.actionHandler_.bind(this));
+    // Wait for amp-selector to finish loading if the current form is using it.
+    this.whenDependenciesReady_().then(() => {
```

Comment:
> esp since we have other listeners that could override the behavior

You mean other `submit` listeners? This would still be very similar situation with blocking on dependencies there are still other listeners.

If you meant other action handlers yeah if some other code install an action listener on the form element these `submit` events will be gone from the queue.